src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -6,13 +6,16 @@
 using System.Linq;
 using System.Security.Claims;
 using System.Text;
+using System.Text.RegularExpressions;
 using Microsoft.Extensions.Caching.Memory;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Authentication.Negotiate;
 
 internal static class LdapAdapter
 {
+    private static readonly Regex DistinguishedNameSeparator = new Regex(@"(?<!\\),");
```

Comment:
Use source generator?

https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators

(I don't have context on the change, but we would prefer statically defined Regex are source generated)

Reply:
I just realized I'm not in the runtime repo, so this may not apply...

Reply:
Done.

Reply:
I suggest the method name should end with "Regex" as this is the emerging convention.

I'll let a domain expert review the correctness of the change itself...